packages/client/src/tlog/verify/index.ts

```diff
@@ -31,6 +38,20 @@ export function verifyTLogEntries(
   bundle: Bundle,
   trustedRoot: sigstore.TrustedRoot,
   options: sigstore.ArtifactVerificationOptions_TlogOptions
+): void {
+  if (bundle.mediaType === BUNDLE_V01_MEDIA_TYPE) {
+    assertBundleV01(bundle);
+    verifyTLogEntriesForBundleV01(bundle, trustedRoot, options);
+  } else {
+    assertBundleLatest(bundle);
+    verifyTLogEntriesForBundleLatest(bundle, trustedRoot, options);
+  }
```

Comment:
This is where we determine which style of verification to apply to the bundle. The `assert*` functions will check for completeness of the bundle and also narrow the type to indicate the presence of either the `inclusionPromise` or `inclusionProof`.